Settings.template.txt
{
    "sparqlEndpoint": "https://dbpedia.org/sparql",
    "baseURI": "http://dbpedia.org",
    "providerType": "GenericSparql",
    "updateEndpoint": "",
    "graphStoreEndpoint": "",
    "repositoryId": "",
    "catalogId": "",
    "username": "",
    "password": "",
    "timeoutMilliseconds": 0,
    "preferJsonResults": false,
    "databaseSupportsBifContains": false,
    "defaultNodeCreationURI": "http://graph2vr.org/newNode#",
    "defaultEdgeCreationURI": "http://graph2vr.org/newEdge#",
    "initialSparqlQueryString": "CONSTRUCT { ?s ?p ?o . } WHERE { ?s ?p ?o . } LIMIT 25",
    "initialSparqlURI": "http://www.w3.org/2002/07/owl#Thing",
    "startWithSingleNode": false,
    "searchOnKeypress": false,
    "queryLoggingEnabled": false,
    "playerHeight": 1.8,
    "autoShowNodeMedia": true,
    "pdbRepresentation": 0,
    "modelNodeSize": 0.75,
    "screenshotWidth": 7680,
    "screenshotHeight": 4320,
    "screenshotDisablePostProcessing": true,
    "screenshotHideControllers": false,
    "imagePredicates": [
        "http://xmlns.com/foaf/0.1/depiction",
        "http://xmlns.com/foaf/0.1/Image",
        "http://xmlns.com/foaf/0.1/thumbnail",
        "http://dbpedia.org/property/photo",
        "http://dbpedia.org/ontology/thumbnail"
    ],
    "modelPredicates": [
        "http://graph2vr.org/model",
        "http://graph2vr.org/mesh"
    ],
    "infopanelPredicates": [
        "http://www.w3.org/1999/02/22-rdf-syntax-ns#type",
        "http://www.w3.org/2000/01/rdf-schema#label"
    ],
    "literalColor": {
        "r": 1.0,
        "g": 0.8,
        "b": 0.2,
        "a": 1.0
    },
    "nodeOwlClassColor": {
        "r": 0.8,
        "g": 0.6,
        "b": 0.8,
        "a": 0.0
    },
    "nodeRdfsClassColor": {
        "r": 0.6666667,
        "g": 0.8,
        "b": 1.0,
        "a": 0.0
    },
    "nodeOwlDatatypeColor": {
        "r": 0.6,
        "g": 0.8,
        "b": 0.4,
        "a": 0.0
    },
    "deprecatedColor": {
        "r": 0.8,
        "g": 0.8,
        "b": 0.8,
        "a": 1.0
    },
    "arrowheadSubclassOfColor": {
        "r": 1.0,
        "g": 1.0,
        "b": 1.0,
        "a": 1.0
    },
    "databaseSetttings": [
        {
            "label": "DBpedia",
            "sparqlEndpoint": "https://dbpedia.org/sparql",
            "baseURI": "http://dbpedia.org",
            "providerType": "GenericSparql",
            "updateEndpoint": "",
            "graphStoreEndpoint": "",
            "repositoryId": "",
            "catalogId": "",
            "username": "",
            "password": "",
            "timeoutMilliseconds": 0,
            "preferJsonResults": false,
            "databaseSupportsBifContains": false,
            "searchOnKeypress": true
        },
        {
            "label": "Local Virtuoso",
            "sparqlEndpoint": "http://localhost:8890/sparql",
            "baseURI": "",
            "providerType": "Virtuoso",
            "updateEndpoint": "",
            "graphStoreEndpoint": "",
            "repositoryId": "",
            "catalogId": "",
            "username": "",
            "password": "",
            "timeoutMilliseconds": 0,
            "preferJsonResults": false,
            "databaseSupportsBifContains": true,
            "searchOnKeypress": false
        },
        {
            "label": "Local GraphDB",
            "sparqlEndpoint": "http://localhost:7200/repositories/example",
            "baseURI": "",
            "providerType": "GraphDB",
            "updateEndpoint": "",
            "graphStoreEndpoint": "",
            "repositoryId": "example",
            "catalogId": "",
            "username": "",
            "password": "",
            "timeoutMilliseconds": 0,
            "preferJsonResults": false,
            "databaseSupportsBifContains": false,
            "searchOnKeypress": false
        },
        {
            "label": "Local AllegroGraph",
            "sparqlEndpoint": "http://localhost:10035",
            "baseURI": "",
            "providerType": "AllegroGraph",
            "updateEndpoint": "",
            "graphStoreEndpoint": "",
            "repositoryId": "example",
            "catalogId": "",
            "username": "",
            "password": "",
            "timeoutMilliseconds": 0,
            "preferJsonResults": false,
            "databaseSupportsBifContains": false,
            "searchOnKeypress": false
        },
        {
            "label": "Local Fuseki",
            "sparqlEndpoint": "http://localhost:3031/ds/sparql",
            "baseURI": "",
            "providerType": "Fuseki",
            "updateEndpoint": "",
            "graphStoreEndpoint": "",
            "repositoryId": "",
            "catalogId": "",
            "username": "",
            "password": "",
            "timeoutMilliseconds": 3600000,
            "preferJsonResults": true,
            "databaseSupportsBifContains": false,
            "searchOnKeypress": false
        },
        {
            "label": "Stardog Example",
            "sparqlEndpoint": "https://example.stardog.cloud:5820",
            "baseURI": "",
            "providerType": "Stardog",
            "updateEndpoint": "",
            "graphStoreEndpoint": "",
            "repositoryId": "catalog",
            "catalogId": "",
            "username": "",
            "password": "",
            "timeoutMilliseconds": 3600000,
            "preferJsonResults": true,
            "databaseSupportsBifContains": false,
            "searchOnKeypress": false
        }
    ],
    "predefinedPredicates": [
        {
            "name": "owl:sameAs",
            "uri": "http://www.w3.org/2002/07/owl#sameAs"
        },
        {
            "name": "skos:exactMatch",
            "uri": "http://www.w3.org/2004/02/skos/core#exactMatch"
        },
        {
            "name": "skos:closeMatch",
            "uri": "http://www.w3.org/2004/02/skos/core#closeMatch"
        },
        {
            "name": "rdfs:seeAlso",
            "uri": "http://www.w3.org/2000/01/rdf-schema#seeAlso"
        }
    ],
    "savedMaximumImageWidth": 512,
    "savedMaximumImageHeight": 512
}


Assets/Scripts/Settings/Settings.cs
/*******************************************************
 * Copyright (C) 2017 Doron Weiss  - All Rights Reserved
 * You may use, distribute and modify this code under the
 * terms of unity license.
 * 
 * See https://abnormalcreativity.wixsite.com/home for more info
 *******************************************************/

using System.Collections.Generic;
using UnityEngine;

namespace Dweiss
{
  [System.Serializable]
  public class DatabaseSetttings
  {
    public string label = "";
    public string sparqlEndpoint = "https://dbpedia.org/sparql";
    public string baseURI = "http://dbpedia.org"; //"https://github.com/PjotrSvetachov/GraphVR/example-graph";
    public string providerType = "GenericSparql";
    public string updateEndpoint = "";
    public string graphStoreEndpoint = "";
    public string repositoryId = "";
    public string catalogId = "";
    public string username = "";
    public string password = "";
    public int timeoutMilliseconds = 0;
    public bool preferJsonResults = false;
    public bool databaseSupportsBifContains = false;
    public bool searchOnKeypress = false;
  }

  [System.Serializable]
  public class PredefinedPredicate
  {
    public string name;
    public string uri;
  }

  [System.Serializable]
  public class Settings : ASettings
  {  // Temporary removed baseclase ASettings
    [Header("--Main settings--")]
    //public string BaseURI = "";
    //"https://github.com/PjotrSvetachov/GraphVR/example-graph";
    // "http://dbpedia.org";

    public string sparqlEndpoint = "https://dbpedia.org/sparql";
    public string baseURI = "http://dbpedia.org"; //"https://github.com/PjotrSvetachov/GraphVR/example-graph";
    public string providerType = "GenericSparql";
    public string updateEndpoint = "";
    public string graphStoreEndpoint = "";
    public string repositoryId = "";
    public string catalogId = "";
    public string username = "";
    public string password = "";
    public int timeoutMilliseconds = 0;
    public bool preferJsonResults = false;
    public bool databaseSupportsBifContains = false;

    public string defaultNodeCreationURI = "http://graph2vr.org/newNode#";
    public string defaultEdgeCreationURI = "http://graph2vr.org/newEdge#";

    // We can start with an single node or a query.
    public string initialSparqlQueryString = "construct {?s ?p ?o}  where  {?s ?p ?o} Limit 100";
    // prefix dbo: <http://dbpedia.org/ontology/> prefix rdfs: <http://www.w3.org/2000/01/rdf-schema#> prefix rdf: <http://www.w3.org/1999/02/22-rdf-syntax-ns#> prefix dbp: <http://dbpedia.org/property/> prefix dbr: <http://dbpedia.org/resource/> construct {     ?mountain rdf:type dbo:Mountain.     ?mountain dbo:elevation ?height.     ?mountain dbp:location ?location }  where {     ?mountain rdf:type dbo:Mountain.     ?mountain dbo:elevation ?height.     ?mountain dbp:location ?location }  ORDER BY DESC (?height)  LIMIT 10
    /*
     prefix dbo: <http://dbpedia.org/ontology/> 
      prefix rdfs: <http://www.w3.org/2000/01/rdf-schema#> 
      prefix rdf: <http://www.w3.org/1999/02/22-rdf-syntax-ns#> 
      prefix dbp: <http://dbpedia.org/property/> 
      prefix dbr: <http://dbpedia.org/resource/> 
      construct { ?mountain rdf:type dbo:Mountain.     
                  ?mountain dbo:elevation ?height.     
                  ?mountain dbp:location ?location.
                  ?mountain <http://graph2vr.org/label> ?label.
                  ?mountain <http://graph2vr.org/image> ?image.
                 }  
      where {     ?mountain rdf:type dbo:Mountain.     
                  ?mountain dbo:elevation ?height.     
                  ?mountain dbp:location ?location. 
                  Optional{?mountain rdfs:label ?label.}
                  Optional{?mountain dbp:depiction ?image.}
      }  
      ORDER BY DESC (?height)  LIMIT 10
    */
    public string initialSparqlURI = "";
    public bool startWithSingleNode = false;
    public bool searchOnKeypress = false;
    public bool queryLoggingEnabled = false;
    public float playerHeight = 1.8f;
    public bool autoShowNodeMedia = true;
    public PdbRepresentation pdbRepresentation = PdbRepresentation.Bonds;
    public float modelNodeSize = 0.75f;
    public int screenshotWidth = 7680;
    public int screenshotHeight = 4320;
    public bool screenshotDisablePostProcessing = true;
    public bool screenshotHideControllers = false;

    public string[] imagePredicates = { "http://xmlns.com/foaf/0.1/depiction", "http://xmlns.com/foaf/0.1/Image", "http://xmlns.com/foaf/0.1/thumbnail", "http://dbpedia.org/property/photo" };
    public string[] modelPredicates = { "http://graph2vr.org/model", "http://graph2vr.org/mesh" };

    public string[] infopanelPredicates = { "http://www.w3.org/1999/02/22-rdf-syntax-ns#type", "http://www.w3.org/2000/01/rdf-schema#label" };

    // VOWL color schema
    // Datatype 	#fc3 	yellow 	rdfs:Datatype, rdfs:Literal
    public Color literalColor = new Color(255, 204, 51);
    // General 	#acf 	light blue 	owl:Class, owl:ObjectProperty (incl. subclasses)
    public Color nodeOwlClassColor = new Color(170, 204, 255);
    // Rdf 	#c9c 	light purple 	rdfs:Class, rdfs:Resource, rdf:Property
    public Color nodeRdfsClassColor = new Color(204, 153, 204);
    // Datatype Property 	#9c6 	light green owl:DatatypeProperty
    public Color nodeOwlDatatypeColor = new Color(153, 204, 102);
    // Deprecated 	#ccc 	light gray 	owl:DeprecatedClass, owl:DeprecatedProperty
    public Color deprecatedColor = new Color(204, 204, 204);
    // Neutral 	#fff 	white   owl:Thing, arrowhead of rdfs:subClassOf
    public Color arrowheadSubclassOfColor = new Color(255, 255, 255);

    public List<DatabaseSetttings> databaseSetttings = new List<DatabaseSetttings>();
    public List<PredefinedPredicate> predefinedPredicates = new List<PredefinedPredicate>();

    public int savedMaximumImageWidth = 512;
    public int savedMaximumImageHeight = 512;

    private void Awake()
    {
      base.Awake ();
      SetupSingelton();
      int savedPdbMode = PlayerPrefs.GetInt("PdbRepresentation", (int)pdbRepresentation);
      pdbRepresentation = System.Enum.IsDefined(typeof(PdbRepresentation), savedPdbMode)
        ? (PdbRepresentation)savedPdbMode : PdbRepresentation.Bonds;
      if (PlayerPrefs.HasKey("AutoShowNodeMedia"))
        autoShowNodeMedia = PlayerPrefs.GetInt("AutoShowNodeMedia") != 0;
    }

    public void SetPdbRepresentation(PdbRepresentation representation)
    {
      if (!System.Enum.IsDefined(typeof(PdbRepresentation), representation)) return;
      pdbRepresentation = representation;
      PlayerPrefs.SetInt("PdbRepresentation", (int)pdbRepresentation);
      PlayerPrefs.Save();
    }

    public void SetAutoShowNodeMedia(bool enabled)
    {
      autoShowNodeMedia = enabled;
      PlayerPrefs.SetInt("AutoShowNodeMedia", enabled ? 1 : 0);
      PlayerPrefs.Save();
    }

    new string name = "";
    #region  Singleton
    public static Settings _instance;

    public static Settings Instance { get { return _instance; } }
    private void SetupSingelton()
    {
      if (_instance != null)
      {
        Debug.LogError("Error in settings. Multiple singletons exists: " + _instance.name + " and now " + this.name);
      }
      else
      {
        _instance = this;
      }
    }
    #endregion



  }
}
